addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java

```diff
@@ -0,0 +1,1145 @@
+                logger.debug("Sleep was interrupted: {}", e.getMessage());
+            } catch (IOException e) {
+                logger.debug("Fatal transport error: {}", e.getMessage());
+            } catch (NullPointerException e) {
+                logger.debug("No valid response:", e.getMessage());
+            } finally {
+                // Release the connection.
+                method.releaseConnection();
+            }
+            return responseBodyB64;
+        }
+    }
+
+    /**
+     * This function does the SEND http communication to the device
+     *
+     */
+    public Integer sendDataToService(String service, byte[] data) {
+        // Create an instance of HttpClient.
+        Integer rCode = null;
+        synchronized (client) {
+            logger.debug("Starting send connection...");
+
+            // Create a method instance.
+            PostMethod method = new PostMethod("http://" + device.getIP4Address() + service);
+
+            // Provide custom retry handler is necessary
+            method.getParams().setParameter(HttpMethodParams.RETRY_HANDLER,
+                    new DefaultHttpMethodRetryHandler(3, false));
+            // Set the right header
+            method.setRequestHeader("Accept", "application/json");
+            method.addRequestHeader("User-Agent", "TeleHeater/2.2.3");
+            method.setRequestEntity(new ByteArrayRequestEntity(data));
+
+            try {
+                rCode = client.executeMethod(method);
+            } catch (IOException e) {
+                logger.debug("Failed to send data {}", e);
+            } finally {
+                // Release the connection.
+                method.releaseConnection();
+            }
+            logger.debug("Returncode: {}", rCode);
+            return rCode;
+        }
+    }
+
+    /**
+     * This function does the decoding for a new message from the device
+     *
+     */
+    public String decodeMessage(byte[] encoded) {
+        String retString = null;
+        byte[] decodedB64 = null;
+
+        decodedB64 = Base64.decodeBase64(encoded);
+        try {
+            /* Check whether the length of the decryptData is NOT multiplies of 16 */
+            if ((decodedB64.length & 0xF) != 0) {
+                /* Return the data */
+                retString = new String(decodedB64, device.getCharSet());
+                return retString;
+            }
+            // --- create cipher
+            final Cipher cipher = Cipher.getInstance("AES/ECB/NoPadding");
+            cipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(device.getCryptKeyPriv(), "AES"));
+            byte[] decryptedData = cipher.doFinal(decodedB64);
+            byte[] decryptedDataWOZP = removeZeroPadding(decryptedData);
+            return (new String(decryptedDataWOZP, device.getCharSet()));
+        } catch (UnsupportedEncodingException | GeneralSecurityException e) {
+            logger.debug("Exception on encoding: {}", e);
+            return null;
```

Comment:
This is not really helping your caller why not just returning an exception instead that is easier to handle for the caller then some null.